DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Ise\WebSecurityBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\EnumNodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder('ise_web_security', 'array');
+
+        if (method_exists($treeBuilder, 'getRootNode')) {
+            $rootNode = $treeBuilder->getRootNode();
+        } else {
+            // BC for symfony/config < 4.2
+            $rootNode = $treeBuilder->root('ise_web_security');
+        }
+
+        $rootNode
+            ->children()
+                ->scalarNode('report_uri')->end()
+                ->arrayNode('coop')
+                    ->children()
+                        ->booleanNode('active')->defaultFalse()->end()
+                    ->end()
+                ->end()
+                ->arrayNode('fetch_metadata')
+                    ->children()
+                        ->booleanNode('active')->defaultFalse()->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+
+        return $treeBuilder;
+    }
+
+    private function getCOOP(): EnumNodeDefinition {
+        $node = new EnumNodeDefinition('coop');
+        $node->values(['none', 'cross-origin', 'same-site', 'same-origin', 'same-origin-allow-popups', 'report-only']);
```

Comment:
Not quite sure I understand this completely. I could change this to a scalar freeform field to leave it to the developer. However this would cause some issues with respect to